src/EditorFeatures/Core.Cocoa/Snippets/AbstractSnippetExpansionClient.cs

```diff
@@ -51,7 +51,7 @@ public AbstractSnippetExpansionClient(IThreadingContext threadingContext, IConte
 
         public abstract IExpansionFunction GetExpansionFunction(XElement xmlFunctionNode, string fieldName);
         protected abstract ITrackingSpan InsertEmptyCommentAndGetEndPositionTrackingSpan();
-        internal abstract Document AddImports(Document document, int position, XElement snippetNode, bool placeSystemNamespaceFirst, CancellationToken cancellationToken);
+        internal abstract Document AddImports(Document document, int position, XElement snippetNode, bool placeSystemNamespaceFirst, bool allowInHiddenRegions, CancellationToken cancellationToken);
```

Comment:
the majority of this PR is just threading this option through *everything.........*